### ISSUE TITLE
Browser doesn't reset upon creation of new wallet 

### ISSUE THREAD

**Issue by ibrahimtaveras00** [MetaMask/metamask-mobile#1660]
**Describe the bug**

When creating a new wallet, browser keeps the same information (favorites + connected dapps) 

**Screenshots**

Seen here = https://recordit.co/xiyrAZjtq5

**To Reproduce**

1. Go to browser
2. Select a dapp and connect
3. Save as favorite
4. Logout and create a new wallet
5. Go back to browser and notice your browser is still in the same state 

**Expected behavior**

Browser should be reset

**Smartphone (please complete the following information):**
 - Device: All
 - OS: All
 - Version All

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by rickycodes** [comment]
do we want to blow out favorites and connected dapps when a user creates or imports a new wallet? I feel like there might be value in keeping the old favorites? /cc @omnat

**Comment by omnat** [comment]
Good point. Agree, there's value in keeping old favorites. After a wallet reset, we should still close the open tabs, as the old 'connected dapps' won't work with the new wallet.

**Comment by AlexJupiter** [comment]
For this to be closed we should do Omna's suggestion above:
>After a wallet reset, we should still close the open tabs, as the old 'connected dapps' won't work with the new wallet.

**Comment by hesterbruikman** [comment]
Needs architectural review fore feasibility cc @sethkfman
